src/guards/roles.guard.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { UserRole } from 'src/entities/role.enum';
+
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+  constructor(private reflector: Reflector) {}
+
+  canActivate(context: ExecutionContext): boolean {
```

Comment:
async-await và viết exception cụ thể hơn về UnauthorizedException không trả về true false

Reply:
return true is acceptable, just not return false